src/ripple/protocol/SOTemplate.h

```diff
@@ -61,50 +75,44 @@ class SOElement
 class SOTemplate
 {
 public:
-    using list_type = std::vector <std::unique_ptr <SOElement const>>;
-    using iterator_range = boost::iterator_range<list_type::const_iterator>;
+    SOTemplate(SOTemplate&& other) = default;
```

Comment:
I'd either explictly delete the copy constructor/assignment operator or add a comment here explaining the reason for the defaulted move constructor (assuming this is the reason).

BTW, I don't see an obvious reason why this is a move-only type, but I'm not advocating for that change now.

Reply:
Thanks for poking at this.  It was not a corner I thought hard about.

I have a habit, when I _am_ thinking about it, of only giving classes capabilities that they need at the moment.  Since we have non-library code adding that capability later is not really a hardship.  And, if the class doesn't have that capability initially, the person who wants that feature can think about why they need it rather an accidentally introducing, say, a copy they weren't thinking about.

That's why, for instance, I was on the side of the Standard Committee that wanted operator
spaceship to be op-in.  I didn't want yet one more thing that I need to remember to explicitly remove from most of the classes I write.

In this case it turns out that the usage model in this pull request doesn't require `SOTemplate` to have _any_ constructor or assignment operator beyond the constructor that takes an `initializer_list`.  So the obvious thing (to me) is to simply remove the capabilities that I don't need: all those constructors and assignment operators.

However my approach does make it more inconvenient for folks who want to do new stuff in the future.  And my approach would never work for library development (where we have no idea how a potential user might want to use a class in the library).

So at this point I would be inclined to explicitly delete the constructors and assignment operators that are not needed by the code base at the moment.  I'd also add a comment saying that they are just not needed and can be enabled if the need arises.  Does that work for you?

Thanks.  And thanks for noting things that look off to you.

Reply:
By default, I'd like our types to regular types, and then remove functionality we want to forbid for good reasons - and document those reasons. A move-only type should be a signal about how that type is meant to be used (like `unique_pointer` say), not a signal that "no one needed that functionality yet".

In this particular case, the type was already move-only, and I'm not proposing changing that in this PR. However, given your reasoning above it sounds like you want to do this for new classes as well, and I do disagree with that. I want our rippled classes to be movable/copyable/assignable/equal comparable (and orderable and hashable when applicable). Even when there is no code using that functionality.

Reply:
I can live with that.  I'm coding for a shared code base, and I recognize that my approach is not the only one.  Thanks.